examples/tutorials/curation/plot_1_automated_curation.py

```diff
@@ -0,0 +1,248 @@
+# We can choose which to load in the ``load_model`` function as follows:
+#
+# .. code-block::
+#    import spikeinterface.curation as sc
+#    model, model_info = sc.load_model(
+#        repo_id = "AnoushkaJain3/curation_machine_learning_models",
+#        model_name= 'noise_neuron_model.skops',
+#    )
+#
+# If you run this locally you will receive an error:
+#
+# .. code-block::
+#
+#     UntrustedTypesFoundException: Untrusted types found in the file: ['sklearn.metrics._classification.balanced_accuracy_score', 'sklearn.metrics._scorer._Scorer', 'sklearn.model_selection._search_successive_halving.HalvingGridSearchCV', 'sklearn.model_selection._split.StratifiedKFold']
+#
+# This is a security warning. Sharing models, with are Python objects, is complicated.
+# We have chosen to use the `skops format <https://skops.readthedocs.io/en/stable/>`_, instead
+# of the common but insecure ``.pkl`` format (read about ``pickle`` security issues
+# `here <https://lwn.net/Articles/964392/>`_). While unpacking the ``.skops`` file, each function
+# is checked. Ideally, skops should recognise all `sklearn`, `numpy` and `scipy` functions and
+# allow the object to be loaded if it only contains these (and no unkown malicious code). But
+# when ``skops`` it's not sure, it raises an error. Here, it doesn't recognise
+# ``['sklearn.metrics._classification.balanced_accuracy_score', 'sklearn.metrics._scorer._Scorer',
+# 'sklearn.model_selection._search_successive_halving.HalvingGridSearchCV',
+# 'sklearn.model_selection._split.StratifiedKFold']``. Taking a look, these are all functions
+# from `sklearn`, and we can happily add them to the ``trusted`` functions to load:
+#
+# .. code-block::
+#
+#     model, model_info = sc.load_model(
+#         model_name = 'noise_neuron_model.skops',
+#         repo_id = "AnoushkaJain3/curation_machine_learning_models",
+#         trusted = ['sklearn.metrics._classification.balanced_accuracy_score', 'sklearn.metrics._scorer._Scorer', 'sklearn.model_selection._search_successive_halving.HalvingGridSearchCV', 'sklearn.model_selection._split.StratifiedKFold']
+#     )
+#
+# As ``skops`` continues to be developed, we hope more of these functions will be :code:`trusted`
+# by default.
+#
+# If you unequivocally trust the model (e.g. if you have created it), you can bypass this security
+# step by passing ``trust_model = True`` to the ``load_model`` function.
+#
+# In general, you should be cautious when downloading ``.skops`` files and ``.pkl`` files from repos,
+# especially from unknown sources.
+#
+#
+# Directly applying a sklearn Pipeline
+# ------------------------------------
+#
+# Instead of using ``HuggingFace`` and ``skops``, you might have another way of receiving a sklearn
+# pipeline, and want to apply it to your sorted data.
+
+from spikeinterface.curation.model_based_curation import ModelBasedClassification
+
+model_based_classification = ModelBasedClassification(sorting_analyzer, model)
```

Comment:
What is this step doing? I don't quite understand the class name vs. what we were doing before, isn't everything model based classification?

Reply:
is this another way of doing `auto_label_units()`? If so, is it feasible / good to combine into one API providing a model (either a model object or string to huggingface model?

Reply:
Yup yup, we can do this using auto_label_units